app/src/main/java/org/mifos/mobile/ui/recent_transactions/RecentTransactionScreen.kt

```diff
@@ -0,0 +1,237 @@
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.pulltorefresh.PullToRefreshContainer
+import androidx.compose.material3.pulltorefresh.rememberPullToRefreshState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifos.mobile.MifosSelfServiceApp
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MFScaffold
+import org.mifos.mobile.core.ui.component.MifosErrorComponent
+import org.mifos.mobile.core.ui.component.MifosProgressIndicator
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.models.Transaction
+import org.mifos.mobile.utils.CurrencyUtil
+import org.mifos.mobile.utils.DateHelper
+import org.mifos.mobile.utils.Network
+import org.mifos.mobile.utils.RecentTransactionUiState
+import org.mifos.mobile.utils.Utils
+
+@Composable
+fun RecentTransactionScreen(
+    viewModel: RecentTransactionViewModel = hiltViewModel(),
+    navigateBack: () -> Unit )
+{
+    val uiState by viewModel.recentTransactionUiState.collectAsStateWithLifecycle()
+    val isRefreshing by viewModel.isRefreshing.collectAsStateWithLifecycle()
+
+    LaunchedEffect(key1 = Unit) {
+        viewModel.loadRecentTransactions(false, 0)
+    }
+
+    RecentTransactionScreenContent(
+        uiState = uiState,
+        navigateBack = navigateBack,
+        retryConnection = { viewModel.loadRecentTransactions(false, 0) },
+        isRefreshing = isRefreshing,
+        onRefresh = { viewModel.refresh() }
+    )
+}
+
+@OptIn( ExperimentalMaterial3Api::class)
+@Composable
+fun RecentTransactionScreenContent(
+    uiState: RecentTransactionUiState,
+    navigateBack: () -> Unit,
+    retryConnection: () -> Unit,
+    isRefreshing: Boolean,
+    onRefresh: () -> Unit) {
+
+    val context = LocalContext.current
+    val pullRefreshState = rememberPullToRefreshState()
```

Comment:
pass isRefreshing and onRefresh in rememberPullToRefreshState

Reply:
The problem is that th implemntation is good, but i don't know bcz of some UI overlap or something, when i swipe down my screen i can't see the progress bar, I have tried both pull refresh in material and nestedScrollConnection in material3